### PR TITLE
Fix a bug when prune is used along with path

### LINKED ISSUE #3131: Test discovery fails when `path` is used with `prune`
How to reproduce

```
- discover:
    - name: external
      how: fmf
      prune: True
      url: https://github.com/eclipse-bluechi
      ref: main
      path: tests
```

Error:
it try to locate resources

`[Errno 2] No such file or directory: '/var/tmp/tmt/run-001/bluechi/plans/tests/discover/external/tests/tests/tier0/...`

but the actual path is 
`/var/tmp/tmt/run-001/bluechi/plans/tests/discover/external/tests/tests/tests/tier0/...`

### PR DESCRIPTION
**Pull request by KwisatzHaderach**
There was a bug found when test pruning is used along with path argument in discover

Fixes #3131 

Pull Request Checklist

* [x] implement the feature
* [x] extend the test coverage
* [x] include a release note

### CODE REVIEW

**Review by psss — CHANGES_REQUESTED** [review]
This fix looks good and I was able to reproduce and verify the issue with the original repository. But the test does not reproduce the problem. It might be related to the use of relative `path` in the plan.

File: tests/discover/prune.sh [hunk 1]
```diff
@@ -8,12 +8,18 @@ rlJournalStart
...
+        rlRun -s "tmt run -i $run discover plans --name plan tests --name test1"
         rlAssertExists "$run/plan/discover/default-0/tests/test1"
         rlAssertNotExists "$run/plan/discover/default-0/tests/test2"
         rlAssertNotExists "$run/plan/discover/default-0/tests/some-file"
     rlPhaseEnd
 
+    rlPhaseStartTest "Discover only with path"
+        rlRun -s "tmt run -i $run discover plans --name nested tests --name test1"
+        rlAssertExists "$run/nested/discover/default-0/tests/test1"
+        rlAssertNotExists "$run/nested/discover/default-0/tests/test2"
```
[the remark this answers is not in the document]
Comment (KwisatzHaderach) [reply tests/discover/prune.sh]: right, works fine with teemtee/tests, even found another issue with .fmf matching

File: tests/discover/prune/nested.fmf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+discover:
+    how: fmf
+    path: nested
```
[the remark this answers is not in the document]
Comment (psss) [reply tests/discover/prune/nested.fmf:4]: With `url` it's now according to the spec ;-)

**Review by psss — APPROVED** [review]
Now looks good. Thanks for fixing this!

**Review by happz — APPROVED** [review]

**Comment by psss** [comment]
The `provision` plan failure is irrelevant:

    Couldn't fetch Fedora releases from oraculum...

Merging.

**Pull request merged**